ark/attest/tsserver/analyzeAssertCall.ts

```diff
@@ -38,3 +42,63 @@ const serializeArg = (
 		typeArgs: context.typeArgs.map((other) => compareTsTypes(arg, other))
 	}
 })
+
+export type Completions = Record<string, string[]> | string
+
+const getCompletions = (attestCall: ts.CallExpression) => {
+	const arg = attestCall.arguments[0]
+	if (arg === undefined) {
+		return {}
+	}
+	const descendants = getDescendants(arg)
+	const file = attestCall.getSourceFile()
+	const text = file.getFullText()
+	let completions: Completions | string = {}
+	const duplicatedPrefixes: string[] = []
+
+	for (const descendant of descendants) {
+		if (ts.isStringLiteral(descendant)) {
+			// descendant.pos tends to be an open quote while d.end tends to be right after the closing quote.
+			// It seems to be more consistent using this to get the pos for the completion over descendant.pos
+			const lastPositionOfInnerString =
+				descendant.end - (/["']/.test(text[descendant.end - 1]) ? 1 : 2)
+
+			const completionData =
+				TsServer.instance.virtualEnv.languageService.getCompletionsAtPosition(
+					file.fileName,
+					lastPositionOfInnerString,
+					undefined
+				)
+
+			const prefix = descendant.text
+			const entries = completionData?.entries ?? []
+
+			if (
+				(entries.length === 1 && entries[0].name !== prefix) ||
+				entries.length > 1
+			) {
+				if (prefix in completions) {
+					duplicatedPrefixes.push(prefix)
```

Comment:
You can just return the error string immediately when this happens rather than gathering the rest of the completions.

Realistically just the string that is duplicated should be enough to make it obvious what went wrong.